docs/source/cpp/flight.rst

```diff
@@ -172,6 +172,110 @@ request/response. On the server, they can inspect incoming headers and
 fail the request; hence, they can be used to implement custom
 authentication methods.
 
+.. _flight-best-practices:
+
+Best practices
+==============
+
+gRPC
+----
+
+When using default gRPC transport options can be passed to it via
+:member:`arrow::flight::FlightClientOptions::generic_options`. For example:
```

Comment:
This is less a best practice, more of a "howto" so I think this should go in the cookbook